src/Sylius/Behat/Page/Shop/Product/IndexPage.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Page\Shop\Product;
+
+use Sylius\Behat\Page\SymfonyPage;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class IndexPage extends SymfonyPage implements IndexPageInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isResourceOnPage($resourceName)
+    {
+        $elements= $this->getDocument()->findAll('css', 'div.header');
+        foreach ($elements as $element) {
+            if ($resourceName === $element->getText()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEmpty()
+    {
+        $isEmpty = strpos($this->getDocument()->find('css', '.message')->getText(), 'There are no products to display');
+        if (false === $isEmpty ) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRouteName()
+    {
+        return 'sylius_shop_partial_product_index';
```

Comment:
I think this should be ``sylius_shop_taxon_show`` and all this should go to ``Taxon\ShowPage``. Because we are indeed viewing a taxon and it's just a detail that we actually display a list of products there. For some taxons we might want to only display child categories and some recommended products and so on. So the "main concept" here is "Showing a Taxon" and product list is particular implementation.

To sum up: Merge these 2 pages in into taxon's show page.